.github/workflows/build-docs-optimized.yml

```
@@ -0,0 +1,298 @@
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+jobs:
+  build:
+    name: Build Documentation (Optimized)
+    runs-on: ubuntu-latest
+    outputs:
+      deploy-url: ${{ steps.deployment.outputs.page_url }}
+      version: ${{ steps.calver.outputs.version }}
+      artifact-name: ${{ steps.upload.outputs.artifact-name }}
```

actionlint (1.7.8)

property "artifact-name" is not defined in object type {artifact_id: string}

(expression)